vnext/Microsoft.ReactNative.Cxx/ReactPromise.cpp

```diff
@@ -105,6 +104,27 @@ bool ReactPromiseBase::TrySetState(State newState) const noexcept {
   return false;
 }
 
+/*static*/ MethodResultCallback ReactPromiseBase::GetRejectResultCallback(
+    std::function<void(ReactError const &)> const &reject) noexcept {
+  return [reject](IJSValueWriter const &outputWriter) noexcept {
+    winrt::com_ptr<JSValueTreeWriter> writer = outputWriter.as<JSValueTreeWriter>();
+    JSValue jsValue = writer->TakeValue();
+    ReactError err{};
+    err.Code = jsValue.AsArray()[0].AsObject()[ErrorMapKeyCode].AsString();
+    err.Message = jsValue.AsArray()[0].AsObject()[ErrorMapKeyMessage].AsString();
+    err.UserInfo = jsValue.AsArray()[0].AsObject()[ErrorMapKeyUserInfo].AsObject().Copy();
```

Comment:
consider a local for `.AsArray()[0].AsObject()` to make this a little less verbose :)